tests/test_arm.py

```diff
@@ -66,8 +65,8 @@ async def test_is_moving(self):
 
     @pytest.mark.asyncio
     async def test_do(self):
-        with pytest.raises(NotImplementedError):
-            await self.arm.do_command({"command": "args"})
+        resp = await self.arm.do_command({"command": "args"})
+        assert resp == {"hello": "world"}
```

Comment:
If you follow my suggestion in `mocks.components`, then the assert should now be `assert responsibility == {"command": {"command": "args"}}`

You could of course save the command we pass as a parameter as a var:

```python3
command = {"command": "args"}
resp = await self.arm.do_command(command)
assert resp == {"command": command}
```